utilities/blob_db/blob_db_impl.cc

```diff
@@ -921,14 +921,17 @@ Status BlobDBImpl::Write(const WriteOptions& opts, WriteBatch* updates) {
       last_file_ = bfile;
       has_put_ = true;
 
-      Slice value = value_unc;
+      Slice value;
       std::string compression_output;
-      if (impl_->bdb_options_.compression != kNoCompression) {
+      if (impl_->bdb_options_.compression == kNoCompression) {
```

Comment:
In lines 924 and 1058, wondering if you can do something like `Slice value = GetFooBarValue(value_unc, compression, ...)` and move the common code into that function?